src/vs/workbench/contrib/chat/browser/contrib/chatDynamicVariables.ts

```diff
@@ -481,13 +486,40 @@ export async function searchFolders(
 		return [];
 	}
 
-	const folderResources = getMatchingFoldersFromFiles(folderResults.results.map(result => result.resource), workspace, pattern);
+	const folderResources = getMatchingFoldersFromFiles(folderResults.results.map(result => result.resource), workspace, segmentMatchPattern);
 	return folderResources;
 }
 
+function fuzzyMatchingGlobPattern(pattern: string): string {
+	if (!pattern) {
+		return '*';
+	}
+	return '*' + pattern.split('').join('*') + '*';
+}
+
+function continousMatchingGlobPattern(pattern: string): string {
+	if (!pattern) {
+		return '*';
+	}
+	return '*' + pattern + '*';
+}
+
+function caseInsensitiveGlobPattern(pattern: string): string {
+	let caseInsensitiveFilePattern = '';
+	for (let i = 0; i < pattern.length; i++) {
+		const char = pattern[i];
+		if (/[a-zA-Z]/.test(char)) {
+			caseInsensitiveFilePattern += `[${char.toLowerCase()}${char.toUpperCase()}]`;
+		} else {
+			caseInsensitiveFilePattern += char;
+		}
+	}
+	return caseInsensitiveFilePattern;
+}
+
```

Comment:
Recommendation: Introduce a `Pattern` class (which also defines how it matches and why it is ok to transform the pattern like that - which wouldn't work for regular expressions for example because of named groups that would be lowercased).